test/e2e/app-dir/global-error/error-in-global-error/error-in-global-error.test.ts

```diff
@@ -0,0 +1,59 @@
+import { nextTestSetup } from 'e2e-utils'
+import { assertHasRedbox } from 'next-test-utils'
+
+describe('app dir - global-error - error-in-global-error', () => {
+  const { next, isNextDev } = nextTestSetup({
+    files: __dirname,
+  })
+
+  it('should be able to use nextjs navigation hook in global-error', async () => {
+    const browser = await next.browser('/')
+    const text = await browser.elementByCss('h1').text()
+    expect(text).toBe('Custom Global Error')
+
+    if (isNextDev) {
+      await assertHasRedbox(browser)
+      await expect(browser).toDisplayRedbox(`
+       {
+         "count": 1,
+         "description": "Error: error in page",
+         "environmentLabel": null,
+         "label": "Unhandled Runtime Error",
+         "source": "app/page.js (7:11) @ Page.useEffect
+       >  7 |     throw new Error('error in page')
+            |           ^",
+         "stack": [
+           "Page.useEffect app/page.js (7:11)",
+         ],
+       }
+      `)
+    }
+  })
+
+  it('should render fallback UI when error occurs in global-error', async () => {
+    const browser = await next.browser('/?error-in-global-error=1')
+    const text = await browser.elementByCss('h2').text()
+    expect(text).toBe(
+      'Application error: a client-side exception has occurred while loading localhost (see the browser console for more information).'
```

Comment:
might need to adjust this matcher to be less sensitive now that we include the hostname in the error, since it won't be `localhost` when deployed